src/metabase/util/log.clj

```diff
@@ -222,12 +223,20 @@
   [context-map & body]
   (macros/case
     :clj `(let [ctx-map# ~context-map
-                ctx-keys# (keys ctx-map#)]
+                ctx-keys# (keys ctx-map#)
+                ;; Store original values before modifying
+                original-values# (into {}
+                                       (keep (fn [k#]
+                                               (when-let [v# (ThreadContext/get (name k#))]
+                                                 [(name k#) v#])))
+                                       ctx-keys#)]
             (try
               (doseq [k# ctx-keys#]
                 (ThreadContext/put (name k#) (str (get ctx-map# k#))))
               ~@body
               (finally
                 (doseq [k# ctx-keys#]
-                  (ThreadContext/remove (name k#))))))
+                  (if-let [original# (get original-values# (name k#))]
```

Comment:
What if we want a literal `false` in the context?

Suggest using `find` instead of `get`.

Reply:
context is always a {String, String} map, so it's always something, but I guess maybe we want an empty string too?

will change to find

Reply:
Even though we stringify what's passed when updating the context, I imagine we'll be lazy and use raw values quite often, so having `false` behave as expected is nice.

Empty string is truthy, so don't think that matters.